.golangci.yml

```diff
@@ -4,13 +4,90 @@ run:
 #   timeout: 5m
 
 linters:
+  # Disable all linters.
+  # Default: false
   disable-all: true
+  # Enable specific linter
+  # https://golangci-lint.run/usage/linters/#enabled-by-default-linters
   enable:
+    - asciicheck
+    - bidichk
+    - bodyclose
+    - contextcheck
+    #o    - cyclop
+    - deadcode
+    - depguard
+    #    - dogsled <- disabled because we would like to participate in the iditerod
+    - dupl
+    - durationcheck
     - errcheck
+    - errname
+    - errorlint
+    - exhaustive
+    #    - exhaustivestruct <- disabled because we don't want to use every element of every struct
+    - exportloopref
+    #    - forbidigo <- forbids fmt.Print* by default
+    - forcetypeassert
+    #    - funlen <- some of our functions are longer than this by necessity
+#    - gci <- questionable utility
+    #    - gochecknoglobals <- disabled because we want some globals
+    #    - gochecknoinits <- disabled because we want to use init()
+    #    - gocognit <- disabled for same reason as gocyclo
+    - goconst
+    - gocritic
+    #    - gocyclo <- disabled because it won't work with how we cosmos
+    - godot
+    #    - godox <- detects TODO/BUG/FIXME and we may wantnthis later but isn't appropriate now
+    #    - goerr113 <- disabled due to lack of comprehension
     - gofmt
-    # - goimports
-    # - golint
-    - maligned
+    - gofumpt
+    - goheader
+    - goimports
+    #    - gomoddirectives <- disables replaces
+    - gomodguard
+    - goprintffuncname
+    - gosec
+    - gosimple
+    - govet
+    #   - ifshort <- seems to be of questionable value
+    - importas
+    - ineffassign
+    #    - ireturn <- disabled because we want to return interfaces
+    #    - lll <- disabled as it is a bit much.  Maybe we have a desired limit?
+    - makezero
+    - misspell
+    - nakedret
+#    - nestif <- ?
+    - nilerr
+    - nilnil
+    #    - nlreturn <- disabled as it doesn't auto-fix something simple and doesn't add anything useful
+    - noctx
+    #    - nolintlint <- disabled because we use nolint in some places
+    - paralleltest
+    - prealloc
+    - predeclared
+    - promlinter
+    - revive
+    - rowserrcheck
+    - sqlclosecheck
+    - staticcheck
+    - structcheck
+    - stylecheck
+    #    - tagliatelle <- disabled for defying cosmos idiom
+    - tenv
+    - testpackage
+    - thelper
+    - tparallel
+    - typecheck
+    - unconvert
+    - unparam
+    - unused
+    - varcheck
+    #    - varnamelen <- disabled because defies idiom
+    #    - wastedassign
+    - whitespace
+#    - wrapcheck
+#    - wsl
 
 issues:
   exclude-rules:
```

Comment:
Can we exclude `client/docs` and have those changes undone?

Reply:
yep, sure can